### PR TITLE
Update Mimetype Dectection

### PR DESCRIPTION
**Pull request by gotmax23**
The directions for setting the mimetype is the readme.md is incorrect and does not work on Nextcloud 16. I have edited.

### CODE REVIEW

**Comment by lil5** [comment]
:-1: This should not work see the schema differences between `/resources/config/mimetypemapping.json` and `/resources/config/mimetypealiases.json`.

The example is of the same scema as `/resources/config/mimetypemapping.json`
```
{
  "kdbx": ["application/x-kdbx"]
}
```

File: README.md [hunk 1]
```diff
@@ -20,10 +20,10 @@ To update to a new version, simply repeat these steps.
 Unfortunately, apps can't declare new mimetypes on the fly. To make
 Keeweb work properly, you need to add a new mimetype in the
 `mimetypemapping.json` file (also see the Nextcloud manual at
-https://docs.nextcloud.com/server/14/admin_manual/configuration_mimetypes/index.html).
+https://docs.nextcloud.com/server/16/admin_manual/configuration_mimetypes/index.html).
 
-To proceed, create the file `/config/mimetypemapping.json` (in the `config/` folder at
-Nextcloud’s root directory; the file should be stored next to the `config.php`
+To proceed, create the file 'mimetypealiases.json' (in the `config/` folder at
```
Comment (lil5) [inline README.md]: Change to `mimetypemapping.json`

**Comment by maurerle** [comment]
**To summerize a little bit:**
I agree to lil5's suggestion.
I can report that it works both ways but based on the usage in nextcloud `mimetypemapping.json` would be the right place for this. (reference: [relevant server code](https://github.com/nextcloud/server/blob/master/lib/private/Files/Type/Detection.php), [Default Mappings](https://github.com/nextcloud/server/blob/master/resources/config/mimetypemapping.dist.json), [Default Aliases](https://github.com/nextcloud/server/blob/master/resources/config/mimetypealiases.dist.json))

So to get everything to work there are two options:
1. setting mimetypealiases
2. setting mimetypemapping (preferred)

When setting up option 1 one needs to run the following two commands:
1. `sudo -u www-data php occ maintenance:mimetype:update-js` (needed to recreate the mimetypealiases located at core/js/mimetypelist.js)
2. `sudo -u www-data php occ maintenance:mimetype:update-db --repair-filecache` (to update the filetype mapping and database.

**When setting up option 2 you only need to run**
`sudo -u www-data php occ maintenance:mimetype:update-db --repair-filecache` (to update the filetype mapping and database.

(The command `sudo -u www-data php occ files:scan --all` does this too, but does a lot of unneeded stuff more.)

Source: [Nextcloud Server docs](https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/occ_command.html#maintenance-commands) and manual testing

This would close #92 , #116 and other mimetype related issues.
Afterwards we could activate the app for general usage with Nextcloud 17.

File: README.md [hunk 2]
```diff
@@ -20,10 +20,10 @@ To update to a new version, simply repeat these steps.
 Unfortunately, apps can't declare new mimetypes on the fly. To make
 Keeweb work properly, you need to add a new mimetype in the
 `mimetypemapping.json` file (also see the Nextcloud manual at
-https://docs.nextcloud.com/server/14/admin_manual/configuration_mimetypes/index.html).
+https://docs.nextcloud.com/server/16/admin_manual/configuration_mimetypes/index.html).
```
Comment (maurerle) [inline README.md]: should be `stable` to point always to the latest documentation

also refer to 
https://docs.nextcloud.com/server/stable/admin_manual/configuration_server/occ_command.html#maintenance-commands

File: README.md [hunk 3]
```diff
@@ -33,9 +33,10 @@ lines:
 }
 ```
 
-After that, run the following command in the root directory of Nextcloud on the server
+After that, run the following commands in the root directory of Nextcloud on the web server
 (if needed, replace `www-data` with the actual user which is used by the webserver):
 
+    sudo -u www-data php occ maintenance:mimetype:update-js
```
Comment (maurerle) [inline README.md]: remove `update-js` and `files:scan` in favor of `sudo -u www-data php occ maintenance:mimetype:update-db --repair-filecache` as stated in my comment

**Comment by kesselb** [comment]
Could you help me testing https://github.com/jhass/nextcloud-keeweb/pull/124? Hopefully that will make the manual change to `mimetypemapping.json` superfluous. Thanks :+1:

**Pull request closed without merging**